src/Worker/Core/Shims/TaskActivityShim.cs

```diff
@@ -34,11 +42,25 @@ public TaskActivityShim(DataConverter dataConverter, TaskName name, ITaskActivit
         string? strippedRawInput = StripArrayCharacters(rawInput);
         object? deserializedInput = this.dataConverter.Deserialize(strippedRawInput, this.implementation.InputType);
         TaskActivityContextWrapper contextWrapper = new(coreContext, this.name);
-        object? output = await this.implementation.RunAsync(contextWrapper, deserializedInput);
 
-        // Return the output (if any) as a serialized string.
-        string? serializedOutput = this.dataConverter.Serialize(output);
-        return serializedOutput;
+        string instanceId = coreContext.OrchestrationInstance.InstanceId;
+        this.logger.ActivityStarted(instanceId, this.name);
+
+        try
+        {
+            object? output = await this.implementation.RunAsync(contextWrapper, deserializedInput);
+
+            this.logger.ActivityCompleted(instanceId, this.name);
```

Comment:
recommend moving this after output serialization - that could fail and then they would see both a completed & failed message.